File: examples/java/Basic.java
import io.greyfoundry.identifold.Identifold;

public final class Basic {
    private Basic() {}

    public static void main(String[] args) {
        var mid = "019d4c72-c910-7a84-b313-53c3ac61a32f";
        var pid = Identifold.publicIdFromMachineId(mid, "order");
        var parsed = Identifold.parsePublicId(pid);
        var roundTrip = parsed.machineId().equals(mid);
        if (!roundTrip) {
            throw new IllegalStateException("MID/PID round trip failed.");
        }
        System.out.printf(
                "{\n  \"mid\": \"%s\",\n  \"namespace\": \"%s\",\n  \"pid\": \"%s\",\n  \"roundTrip\": true\n}%n",
                mid, parsed.namespace(), pid);
    }
}

File: examples/ruby/basic.rb
require "identifold"
require "json"

mid = "019d4c72-c910-7a84-b313-53c3ac61a32f"
pid = Identifold.public_id_from_machine_id(mid, "order")
parsed = Identifold.parse_public_id(pid)
round_trip = parsed[:machineId] == mid
raise "MID/PID round trip failed." unless round_trip

puts JSON.pretty_generate(
  mid: mid,
  namespace: parsed[:namespace],
  pid: pid,
  roundTrip: round_trip
)
